app/src/main/AndroidManifest.xml

```diff
@@ -4,6 +4,7 @@
     package="org.gophillygo.app">
 
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
```

Comment:
The docs are a bit confusing here; the written docs say this (emphases mine):
> You declare a need for foreground location when your app requests **either** the [ACCESS_COARSE_LOCATION](https://developer.android.com/reference/android/Manifest.permission#ACCESS_COARSE_LOCATION) permission **or** the [ACCESS_FINE_LOCATION](https://developer.android.com/reference/android/Manifest.permission#ACCESS_FINE_LOCATION) permission, as shown in the following snippet:


And then the snippet says:
```xml
<manifest ... >
  <!-- Always include this permission -->
  <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />

  <!-- Include only if your app benefits from precise location access. -->
  <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
</manifest>
```

Anyway, it sounds reasonable to me that we trust the code snippet over the written docs, but I am curious whether you encountered an error or some other behavior that caused you to add this?

Reply:
I was referencing the Android developer docs pages on the [upgrade](https://developer.android.com/google/play/requirements/target-sdk#pre12) and on [requesting location access at runtime](https://developer.android.com/training/location/permissions#request-location-access-runtime) on this upgrade, which say:
>On Android 12 (API level 31) or higher, users can request that your app retrieve only approximate location information, even when your app requests the [ACCESS_FINE_LOCATION](https://developer.android.com/reference/android/Manifest.permission#ACCESS_FINE_LOCATION) runtime permission.
>
> To handle this potential user behavior, don't request the ACCESS_FINE_LOCATION permission by itself. Instead, request both the ACCESS_FINE_LOCATION permission and the [ACCESS_COARSE_LOCATION](https://developer.android.com/reference/android/Manifest.permission#ACCESS_COARSE_LOCATION) permission in a single runtime request.

This is in line with the errors I saw while testing the app, which did not allow me to only request only `ACCESS_FINE_LOCATION`.